flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/util/HadoopUtils.java

```diff
@@ -112,6 +112,27 @@ public static Configuration getHadoopConfiguration(org.apache.flink.configuratio
 		return result;
 	}
 
+	public static boolean isCredentialsConfigured(boolean useTicketCache) throws Exception {
+		UserGroupInformation loginUser = UserGroupInformation.getCurrentUser();
```

Comment:
The previous code used `getLoginUser()` (in `HadoopModule`) and `getCurrentUser()` (in `YarnClusterDescriptor), is it ok to only use `getCurrentUser()` now or should we maybe add the UGI as a parameter?

Reply:
Hmm. I might be zeroing in on this problem. Although the [documentation](https://hadoop.apache.org/docs/r2.8.3/api/org/apache/hadoop/security/UserGroupInformation.html) is pretty vague --> seems like `getLoginUser` returns the actual user logged in via keytab or ticket; and `getCurrentUser` return the actual user (login, or the impersonated user via delegation token).

In short, say I am logged in as userA and then use a `UGI.doAs` section as a proxy userB. inside the `doAs` section. getLoginUser returns userA and getCurrentUser returns userB.. This might be the cause of the delegation token based security problem

Reply:
So it seems we should use `getCurrentUser` here?

Reply:
It's a bit tricky after a second look. sometimes Flink actually expects the actual "logged in" user with valid keytab/web-auth so that it can generate additional DT. I am thinking of opening another JIRA to address it as a separate issue; meanwhile I will keep this PR solely fixing the DT error check. what do you think?